### PR TITLE
Example app and google pay availability improvements 

### PR DESCRIPTION
**Pull request by Robert-SD**

### CODE REVIEW

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayAvailabilityChecker.kt [hunk 1]
```diff
@@ -5,40 +5,52 @@ import androidx.fragment.app.FragmentActivity
...
                     )
                 )
             )
         } else {
-            googlePaySetupCallback(Result.failure(Exception("Google pay not available")))
+            googlePaySetupCallback?.invoke(Result.failure(Exception("Google pay not available")))
         }
     }
 
     fun checkGooglePayAvailability(
```
Comment (descorp) [inline android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayAvailabilityChecker.kt:36]: Minor:

How about just "checkAvailability" and "onResult" ? I would not expect `GooglePayAvailabilityChecker` to be able to check availability of something else but GooglePay

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt [hunk 2]
```diff
@@ -26,6 +26,7 @@ class GooglePayComponentManager(
     private val assignCurrentComponent: (ActionHandlingComponent?) -> Unit,
 ) {
     private var googlePayComponent: BaseGooglePayComponentWrapper? = null
+    private val googlePayAvailabilityChecker = GooglePayAvailabilityChecker(activity)
 
     fun isGooglePayAvailable(
```
[the remark this answers is not in the document]
Comment (descorp) [reply android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt:31]: Also, lines 38-47 should be part of `GooglePayAvailabilityChecker` ? 🤔

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt [hunk 3]
```diff
@@ -81,12 +82,14 @@ class GooglePayComponentManager(
             instantPaymentComponentConfigurationDTO.mapToCheckoutConfiguration()
```
Comment (descorp) [inline android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayComponentManager.kt:82]: This line is duplicated with `setupGooglePaySessionComponent `

**Review by descorp — DISMISSED** [review]

File: android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayAvailabilityChecker.kt (hunk 1 above)
Comment (Robert-SD) [reply android/src/main/kotlin/com/adyen/checkout/flutter/components/googlepay/GooglePayAvailabilityChecker.kt:36]: After a second thought, using a dedicated class for only one method is maybe too much. The manager can implement the onAvailabilityResult and that would make the GooglePayAvailabilityChecker completely redundant.

**Review by descorp — APPROVED** [review]

**Pull request merged**